daemon/health.go

```diff
@@ -81,6 +81,11 @@ func (p *cmdProbe) run(ctx context.Context, d *Daemon, cntr *container.Container
 	execConfig.Privileged = false
 	execConfig.User = cntr.Config.User
 	execConfig.WorkingDir = cntr.Config.WorkingDir
+	// health checks don't get the full 10 seconds timeout to stop gracefully.
+	execConfig.StopTimeout = 500 * time.Millisecond
+	if cntr.Config.Healthcheck.StopTimeout != 0 {
+		execConfig.StopTimeout = cntr.Config.Healthcheck.StopTimeout
+	}
```

Comment:
Wondering why you added this back in your last push; is this only to make the test run faster (50 ms instead of 500)? Otherwise we can omit the changes to the `Healthcheck` type

Reply:
Gave it a quick try, and it seems to work, but let me know if you had a specific reason for this 😅 https://github.com/ndeloof/docker/pull/3

Reply:
yes, goal is to keep test fast. Not being able to control timing demonstrate lack of design for testability imho

Reply:
Not everything has to be configurable; we could make the fixed value shorter of course; not sure if saving half a second for a test warrants a change to an exported struct though (if it's only used in a test)